### ISSUE TITLE
No microphone on Chromebook

### ISSUE THREAD

**Issue by martinpopel** [ufal/charles-translator-android#11]
I've install the Android app on my old chromebook (Chrome OS 100.0.4896.82, Acer N15Q8). Everything (including TTS) seems to work, but there is no microphone icon and thus no ASR. The settings does not show anything about ASR (or speech-to-text), just "Data processing" and "Prefer network text-to-speech".

**Comment by krabatom** [comment]
ASR is part of Google Play Services. Majority android phone have it integrated to system, but there are some exceptions like some Huawei phones, custom roms etc. And it looks like Chromebook doesn't have it. I can only detect, that ASR is not detected, but I can't recommend you to install Google Play Services to your device, because there is usually some reason to not have Google Play Services. Do you have Google Play, on your Chrome book?

**Comment by martinpopel** [comment]
Yes, I have Google Play on my Chromebook and I have also "Speech Services by Google" (last updated Mar 23, 2022) installed.
I am OK with closing this issue, if there is nothing else I can do to help with testing.

**Comment by krabatom** [comment]
Speech Services by Google is only for TTS and it is separate app. ASR is different api connected with Google Play Services. Maybe leave it open, I will later try find more about it. And if you use Google Keyboard or anything else (Google Search etc.) with ASR, is it working?

**Comment by martinpopel** [comment]
I am sorry for my ignorance: so "Google Play" is something different than "Google Play Services"? And can I install "Google Play Services" from "Google Play" or how do I detect if it is already installed?
I've never used ASR on the Chromebook. Now, I've installed GBoard, but it says "Step 1: Select Gboard in your Language & input settings", but when I click "Add input methods" in the Chromebook settings, "Gboard" is not offered.
Android apps were added to ChromeOS just a few years ago and I guess the integration is not ideal in old Chromebooks, so maybe there is no good solution.

**Comment by krabatom** [comment]
That's ok :-) Google is funny in naming. They have Google Play, Google Pay, Google Play Services,...

Google Play - it is App store
Google Play Services - hiddeen system app with services+api, that can use other apps, like Google Auht, Google Pay, Google Maps, push notifications etc.

You can find Google Play Services in app manager, but is is system app. In my phone I have option "show system app". Or you can try to open it in Google Play https://play.google.com/store/apps/details?id=com.google.android.gms

**Comment by martinpopel** [comment]
> https://play.google.com/store/apps/details?id=com.google.android.gms

Thanks. When I click on this link on my Chromebook, it seems it is already installed (there is a green button "Deactivate").

**Comment by krabatom** [comment]
Ok, thank you. I will later look at it, why the google api say, that ASR is not available.

**Comment by krabatom** [comment]
Zdeněk Kasner discover, that he must instal https://play.google.com/store/apps/details?id=com.google.android.googlequicksearchbox to have microphone. Can you try it?

**Comment by martinpopel** [comment]
I cannot install com.google.android.googlequicksearchbox on my Chromebook. I get: "Your device isn't compatible with this version."
Google recommends me a similar app: https://play.google.com/store/apps/details?id=com.google.android.apps.searchlite. I've installed it and the mic works there. Afterwards, I install our Translator 0.9.17, but the mic is still not there.

**Comment by krabatom** [comment]
OK, thank you for testing.
